dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+      result += command.toDart();
+    result += '${kIndent*3}],\n';
+    result += '${kIndent*2}),\n';
+    return result;
+  }
+}
+
+/// Represents the animation of a single path command.
+class PathCommandAnimation {
+  const PathCommandAnimation(this.type, this.points);
+
+  /// The command type.
+  final String type;
+
+  /// A matrix with the command's points in different frames.
+  ///
+  /// points[i][j] is the i-th point of the command at frame j.
+  final List<List<Point<double>>> points;
+
+  @override
+  String toString() {
+    return 'PathCommandAnimation(type: $type, points: $points)';
+  }
+
+  String toDart() {
+    String dartCommandClass;
+    switch (type) {
+      case 'M':
+        dartCommandClass = '_PathMoveTo';
+        break;
+      case 'C':
+        dartCommandClass = '_PathCubicTo';
+        break;
+      case 'L':
+        dartCommandClass = '_PathLineTo';
+        break;
+      case 'Z':
+        dartCommandClass = '_PathClose';
+        break;
+      default:
+        throw new Exception('unsupported path command: $type');
+    }
+    String result = '${kIndent*4}const $dartCommandClass(\n';
+    for (List<Point<double>> pointFrames in points) {
+      result += '${kIndent*5}const <Offset> [\n';
+      for (Point<double> point in pointFrames)
+        result += '${kIndent*6}const Offset(${point.x}, ${point.y}),\n';
+      result += '${kIndent*5}],\n';
+    }
+    result +='${kIndent*4}),\n';
+    return result;
+  }
+}
+
+/// Interprets some subset of an SVG* file.
+///
+/// Recursively goes over the SVG tree, applying transforms and opacities,
+/// and build a FrameData which is a flat representation of the paths in the SVG
+/// file, after applying transformations and converting relative coordinates to
+/// absolute.
+///
+/// * Note that this does not support the SVG specification, but is just built to
+/// support SVG files exported by a specific tool the motion design team is
+/// using.
+FrameData interpretSvg(String svgFilePath) {
+  final File file = new File(svgFilePath);
+  if (!file.existsSync()) {
+    throw new ArgumentError('$file does not exist');
+  }
+  final String fileData = file.readAsStringSync();
+  final XmlElement svgElement = _extractSvgElement(xml.parse(fileData));
+  final double width = parsePixels(_extractAttr(svgElement, 'width')).toDouble();
+  final double height = parsePixels(_extractAttr(svgElement, 'height')).toDouble();
+
+  final List<SvgPath> paths =
+    _interpretSvgGroup(svgElement.children, new _Transform());
+  return new FrameData(new Point<double>(width, height), paths);
+}
+
+List<SvgPath> _interpretSvgGroup(List<XmlNode> children, _Transform transform) {
+  final List<SvgPath> paths = <SvgPath>[];
+  for (XmlNode node in children) {
+    if (node.nodeType != XmlNodeType.ELEMENT)
+      continue;
+    final XmlElement element = node;
+
+    if (element.name.local == 'path') {
+      // TODO(amirh): convert relative commands to absolute
```

Comment:
if we intend to support relative commands, we should do this.
otherwise, just document that we don't support those, and remove them from the list of supported commands.

Reply:
(such that it fails when relative commands are found)

Reply:
Oops forgot to remove this TODO, we actually already handle relative commands